### ISSUE TITLE
Not working with Debian stretch 

### ISSUE THREAD

**Issue by mazanj** [qTox/qTox#6362]
OS:Debian stretch
Qtox: version 1.16.3-2

it was running OK under Debiab buster but stopped working after the upgrade. I encounteed and solved a few missing libraries errors, but stumbled upon the missing library libsodium.so.18.
I tried to update Qtox using apt and toxcore2 has been installed, with no result
I checked installed libsodium version, ist ver23 and I still have the missing library error.

How do I have qTox working again?

**Comment by anthonybilinski** [comment]
Packages installed from Debian repos will be updated anytime any of their used libraries are updated. It sounds like what's probably happening is that you're running qTox that has been locally compiled and linked again old system libraries. The libraries have since been updated, so now the locally compiled qTox can't find them. When installing qTox from the repos, it gives you a second copy of qTox, but is still default opening the one that you've built locally.

To check this hypothesis, you could run: `which qtox`. If it's not at /usr/bin/qtox and instead at /usr/local/bin/qtox, that's likely what's happening. If that's the case, you can either recompile and re-install qtox to cause it to re-link to your new system libraries, or you can remove the locally built version, or call /usr/bin/qtox manually to run the system version which is linked to the system libraries.

Note that if your locally compiled version is newer than the version in Debian repos, you might be unable to read past history - but no existing history will be lost. New history will not be saved if it fails to open your history database.

In general you shouldn't have to install specific versions of libraries, so if what I described is what's happening, you should be safe to update, if downgraded through apt, or remove, if downloaded outside of apt, specific versions you've installed.

**Comment by mazanj** [comment]
The hypothesis did not verify; qtox is at  /usr/local/bin/qtox and still give the same error message.  Reinstalling from the repo doesn't do any good.. My real concern is keeping my tox ID

**Comment by anthonybilinski** [comment]
That matches with what I was thinking, but let me try to explain again. You have two qTox's installed on your system:
1) /usr/local/bin/qtox
* Compiled and installed locally (e.g. git clone https://github.com/qtox/qtox.git, cmake ., make, sudo make install)
* Linked to libraries that were on your system at the time of compilation.
* Not updated since when you last ran make, likely on buster.
* Effectively linked to libraries on buster that no longer exist because they've been updated on stretch.
2) /usr/bin/qtox
* Compiled by debian maintainers.
* Installed by apt and inked to the version of libraries that apt is providing you.
* Updated together with any libraries that are updated by apt.

When you updated to stretch, apt updated a bunch of libraries on your system to newer versions, and also would have updated 2), but 1) is unknown by apt and wasn't re-installed by you manually, so is stuck pointing to invalid places.

When you open "qTox" it will default to 1),because /usr/local/bin is earlier in your $PATH variable by default than /usr/bin/.

When you re-install through apt, it's only re-installing 2), which has probably been working fine the whole time, but 1) still needs to be re-linked to the new libraries on your system.

There are a few options, you can do one of the following:
1) Remove /usr/local/bin/qtox so that /usr/bin/qtox is opened by default instead.
2) Re pull, re-compile, and re `make install` your source version of qtox, to update the version at /usr/local/bin
3) Call /usr/bin/qtox explicitly, instead of "qtox" which will default to /usr/local/bin/qtox

In any of these cases your Tox ID and chat logs are saved in `/home/<username>/.config/tox/<qTox username>.tox` and `<qTox username.db>`, both of which won't be removed unless done so explicitly. You can make copies of those files to another folder to be extra sure.

**Comment by mazanj** [comment]
Sorry for a typo that led to the wrong direction. Here are what I got:
a)  which qtox  --> /usr/bin/qtox   (I had this wrong in my post)
b)  ls /usr/local/bin/q*  --> No such file or directory
c)  /usr/bin/qtox  --> /usr/bin/qtox: error while loading shared libraries: libsodium.so.18: cannot open shared object file: No such file or directory

**Comment by anthonybilinski** [comment]
Sorry for the delay, I was a bit stumped.

Checking now, Debian Stretch doesn't have qTox in its repos, Buster and Bullseye do though. That means this qTox at /usr/bin/qtox won't be updated from the Debian repos when it updates your other libs, which is why this problem is still happening like it would if you had a locally compield qTox at /usr/local/bin/qtox.

Any idea where this would be from? Possibly a PPA? If so, the PPA seems out of date, and for the record we don't host any PPAs ourselves (anymore).

So to resolve the issue, you could:
1. Update Debian to Buster or Bullseye, and install qTox from the official repos.
2. [Compile qTox](https://github.com/qTox/qTox/blob/master/INSTALL.md) yourself, which will link against the versions of libraries that you have installed. Note that when you update those libraries through updating your system, you'll need to re-compile and re-install qTox as well.
3. Use the AppImage version of qTox, since it bundles its own libraries: https://github.com/qtox/qtox/releases/tag/v1.17.3

**Comment by anthonybilinski** [comment]
Pretty confident my last comment explains what was happening, I'll close the issue off but if you have further issues feel free to comment.
